rclcpp/test/rclcpp/executors/test_multi_threaded_executor.cpp

```diff
@@ -86,9 +86,11 @@ TEST_F(TestMultiThreadedExecutor, timer_over_take) {
         double diff = static_cast<double>(std::abs((now - last).nanoseconds())) / 1.0e9;
         last = now;
 
-        if (diff < PERIOD - TOLERANCE) {
+        double diff_exp =
+          static_cast<double>(PERIOD) - static_cast<double>(TOLERANCE);
+        if (diff < diff_exp) {
           executor.cancel();
-          ASSERT_GT(diff, PERIOD - TOLERANCE);
+          ASSERT_GT(diff, diff_exp);
         }
```

Comment:
although this is okay to change, but may i ask why? did we have an problem? as far as i see, PERIOD and TOLERANCE are declared with `constexpr double`, which will be used as double at runtime in this case.

Reply:
+1
The changes to this file doesn't make sense to me

Reply:
You are both absolutely right: this change doesn't make any sense :) I chalk it all up to a little burnout and a great desire to see all tests passing. I made this change when the test failed on a local test machine I was using, and convinced myself I had "fixed it" when instead it was a flaky test on that machine, and it just happened to pass after I made the change (but later failed during another run over the weekend).

I'll revert the change. I have also never seen this test fail on the official build machines so I'm leaning towards blaming the failure on the test machine being less powerful than then one on ci.ros2.org (and thus needing some greater tolerance and/or period). I haven't validated this hypothesis though, but I can dig further.

Reply:
Changes reverted by [84c9fcd](https://github.com/ros2/rclcpp/pull/1574/commits/84c9fcd605a9d4a8a0e4314a85aeeb8c77dec31d).